### ISSUE TITLE
Cleanslate inserting &nbsp; character for single spaces

### ISSUE THREAD

**Issue by joshrcook** [wvuweb/cleanslate-cms#120]
## Steps to reproduce the issue
1. Go to a page with an editable region
2. Click inside the editable region.
3. Press the backspace key, then the spacebar.
4. Open the code editor and notice that the two words are now joined by a &nbsp; character
## Results

It produces a &nbsp; character
## Expected results

For spaces that are only 1 character long, it should just produce a standard space
### Environment
- Mac OS X
- Chrome version 39.0.2171.95 (64-bit)
  ![screen shot 2015-01-29 at 4 21 53 pm](https://cloud.githubusercontent.com/assets/2475205/5966505/0bd52ac0-a7d3-11e4-8588-5c67ceb46634.png)

**Comment by dmolsen** [comment]
Making some notes so I don't forget:
- Safari 8 isn't affected
- this is an [open issue](https://code.google.com/p/chromium/issues/detail?id=310149) for the Chrome team that affects _text_ nodes. Maybe Mercury is doing something here that causes those to be text nodes.
- `whitespace: pre-wrap` feels like an amazingly heavy handed fix but it's [actually recommended](https://developers.whatwg.org/editing.html#best-practices-for-in-page-editors) that `contenteditable` regions be set this way. issue here is that we have a lot of content saved the other way and not sure how it actually affects anyone trying to "indent" via whitespace.

This will require some discussion and testing.

**Comment by dmolsen** [comment]
And for the record. My test was:
1. Go to a page with an editable region
2. Click inside the editable region _between two words_.
3. Press the backspace key _to delete the space_, then _press_ spacebar _to re-add_.
4. Open the code editor and notice that the two words are now joined by a `&nbsp;`  character

**Comment by joshrcook** [comment]
It might be worth mentioning that the article you are referring to in reference to `white-space: pre-wrap` doesn't specifically refer to the &nbsp; character, but instead refers to the &blank; character.  That particular CSS rule doesn't help this particular problem.

I also noted that when creating a new paragraph within an editable region, pressing the spacebar doesn't insert a &nbsp; character, but rather just a space.  Even when pressing the spacebar 3 times in succession between words, it inserts this code: `word &nbsp; word`.  So I would say that while typing a paragraph into CleanSlate, it does operate as expected.  It is just when doing the action referred to that it doesn't handle it well.

**Comment by dmolsen** [comment]
@joshrcook -

FWIW, before I wrote up my results I had played around with adding `white-space: pre-wrap` to the `contenteditable` region in Clean Slate. After adding it a delete followed by inserting a space worked the way you wanted. Though multiple spaces ended up just being multiple spaces rather than `&nbsp; &nbsp; &nbsp;`.

I also quickly tested a vanilla `contenteditable` region in Safari and Chrome. Neither had the behavior reported. On the other hand, both the [Mercury Editor demo](http://jejacks0n.github.io/mercury/) and another `contenteditable` editor [Scribe](http://guardian.github.io/scribe/) show the reported behavior. 

At this point I would call your report an unfortunate "expected behavior." Since it happens in both Scribe and Mercury I'm assuming it's a mix of a browser issue and `contenteditable` editor issue and it's out of our power to fix. We might be able to do the `white-space: pre-wrap` styling but, again, I want to run more tests on how the output would work.
